src/test/java/org/springframework/data/elasticsearch/core/ReactiveElasticsearchTemplateIntegrationTests.java

```diff
@@ -1098,6 +1100,30 @@ void shouldReturnExplanationWhenRequested() {
 				})
 				.verifyComplete();
 	}
+
+
+	@Test // #1646
+	@DisplayName("should return info of all indices")
+	void shouldReturnInformationListOfAllIndices() {
+		String indexName = "index-for-reactive-test";
+		IndexCoordinates index = IndexCoordinates.of(indexName);
+
+		// create the index
+		template.indexOps(index).create().block();
+
+		template.indexOps(index)
+				.getInformation()
+				.as(StepVerifier::create)
+				.consumeNextWith(indexInformation -> {
+						assertThat(indexInformation.getName()).isNotNull();
+						assertThat(indexInformation.getName()).isInstanceOf(String.class);
+						assertThat(indexInformation.getMappings()).isInstanceOf(org.springframework.data.elasticsearch.core.document.Document.class);
+						assertThat(indexInformation.getSettings()).isInstanceOf(org.springframework.data.elasticsearch.core.document.Document.class);
+						assertThat(indexInformation.getAliases()).isInstanceOf(Set.class);
+				})
+				.verifyComplete();
+	}
+
```

Comment:
see my comment for the non-reactive code in regards to checking the returned data

Reply:
Is there a file/helper where i could put the following as a static method instead of copy/pasting the same lines?

```java

	private void assertIndexInformationIsCorrect(IndexInformation indexInformation, String indexName) {
		System.out.println(indexInformation.getMappings());
		assertThat(indexInformation.getSettings().get("index.number_of_shards")).isEqualTo("1");
		assertThat(indexInformation.getSettings().get("index.number_of_replicas")).isEqualTo("0");
		assertThat(indexInformation.getSettings().get("index.analysis.analyzer.emailAnalyzer.type")).isEqualTo("custom");

		assertThat(indexInformation.getMappings().get("properties.text.type")).isEqualTo("text");
		assertThat(indexInformation.getMappings().get("properties.text.analyzer")).isEqualTo("synonym_analyzer");

		assertThat(indexInformation.getName()).isEqualTo(indexName);
		assertThat(indexInformation.getMappings()).isInstanceOf(org.springframework.data.elasticsearch.core.document.Document.class);
		assertThat(indexInformation.getSettings()).isInstanceOf(org.springframework.data.elasticsearch.core.document.Document.class);
		assertThat(indexInformation.getAliases()).isInstanceOf(List.class);
	}
```